### PR TITLE
Add coverage support

### PR DESCRIPTION
**Pull request by dagwieers**
This PR includes:
- Add codecov coverage support
- Add missing pieces to xbmc and xbmcgui stubs
- Various fixes to build/test system
- Enable Python 3 testing
- Move plugin entry point into resources/lib/
- Add routing tests
- Include Git requirements for routing and inputstreamhelper

### CODE REVIEW

File: addon.xml [hunk 1]
```diff
@@ -10,9 +10,10 @@
         <import addon="script.module.routing" version="0.2.0"/>
         <import addon="script.module.inputstreamhelper" version="0.3.4"/>
     </requires>
-    <extension point="xbmc.python.pluginsource" library="main.py">
+    <extension point="xbmc.python.pluginsource" library="resources/lib/main.py">
```
[the remark this answers is not in the document]
Comment (dagwieers) [reply addon.xml]: Ok, I reverted it back.

File: test/xbmcaddon.py [hunk 2]
```diff
@@ -28,7 +32,7 @@ def getLocalizedString(msgctxt):
         for entry in PO:
             if entry.msgctxt == '#%s' % msgctxt:
                 return entry.msgstr or entry.msgid
-        return 'vtmtest'
+        return 'vrttest'
```
Comment (michaelarnauts) [inline test/xbmcaddon.py]: Should be vtmtest

Comment (dagwieers) [reply test/xbmcaddon.py]: I should probably change it to something more generic, yet recognizable.
So I don't have to update these things :-/

File: test/run.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+xbmcplugin = __import__('xbmcplugin')
+xbmcvfs = __import__('xbmcvfs')
+
+if len(sys.argv) <= 1:
+    print("%s: URI argument missing\nTry '%s plugin://plugin.video.vtm.go/recent/1' to test." % (sys.argv[0], sys.argv[0]))
+    sys.exit(1)
+
+# Also support bare paths like /recent/2
+if not sys.argv[1].startswith('plugin://'):
+    sys.argv[1] = 'plugin://plugin.video.vrt.nu' + sys.argv[1]
```
Comment (michaelarnauts) [inline test/run.py]: vrt.nu :)

**Pull request merged**